scripts/code_style.py

```diff
@@ -119,13 +119,31 @@ def check_style_is_correct(src_file_list: List[str]) -> bool:
 
         # Uncrustify makes changes to the code and places the result in a new
         # file with the extension ".uncrustify". To get the changes (if any)
-        # simply diff the 2 files.
-        diff_cmd = ["diff", "-u", src_file, src_file + ".uncrustify"]
-        cp = subprocess.run(diff_cmd, check=False)
+        # simply diff the 2 files
+        if use_git_diff:
+            cmp_cmd = ["cmp", "-s", src_file, src_file + ".uncrustify"]
+        else:
+            cmp_cmd = ["diff", "-u", src_file, src_file + ".uncrustify"]
+            
+        cp = subprocess.run(cmp_cmd, check=False)
 
         if cp.returncode == 1:
-            print(src_file + " changed - code style is incorrect.")
             style_correct = False
+            
+            if use_git_diff:
+                if use_pager:
```

Comment:
I can see the point in wanting git to do the diffing, but why a separate option to control paging, rather than rely on the user's general git paging preferences?

Reply:
What I was trying to do was to best emulate the behaviour of diff with git diff, and that means no pager. The problem with this is that I also wanted to use the pager, as that is something I am using to colour my diffs. Admittedly we could leave this up to the user .gitconfig, I was just trying to be more user friendly to those people who might not know that!

Reply:
Ah I see, that makes sense. In that case I think I rescind my point about `--pager` vs `--no-pager`. We want a script that in its default mode just runs and doesn't wait for the user to press `q`.

Reply:
I still don't understand why this script would have a pager option, rather than rely on git configuration and environment variables. If you prefer a pager, wouldn't you prefer to not have to pass additional options? And if you don't want a pager, wouldn't you already have configured git not to use one?